File: src/main/java/com/freya02/bot/docs/CachedMethod.java
package com.freya02.bot.docs;

import com.freya02.docs.SeeAlso;
import net.dv8tion.jda.api.entities.MessageEmbed;

import java.util.List;

public final class CachedMethod {
	private final MessageEmbed methodEmbed;
	private final List<SeeAlso.SeeAlsoReference> seeAlsoReferences;

	public CachedMethod(MessageEmbed methodEmbed,
	                    List<SeeAlso.SeeAlsoReference> seeAlsoReferences) {
		this.methodEmbed = methodEmbed;
		this.seeAlsoReferences = seeAlsoReferences;
	}

	public MessageEmbed getMethodEmbed() {
		return methodEmbed;
	}

	public List<SeeAlso.SeeAlsoReference> getSeeAlsoReferences() {
		return seeAlsoReferences;
	}
}

File: doxxy-bot/src/main/kotlin/dev/freya02/doxxy/bot/docs/cached/CachedMethod.kt
package dev.freya02.doxxy.bot.docs.cached

import dev.freya02.doxxy.bot.docs.index.DocIndex
import dev.freya02.doxxy.bot.docs.metadata.ImplementationIndex
import dev.freya02.doxxy.docs.data.SeeAlso.SeeAlsoReference
import net.dv8tion.jda.api.entities.MessageEmbed

class CachedMethod(
    override val docIndex: DocIndex,
    val className: String,
    val signature: String,
    override val embed: MessageEmbed,
    override val seeAlsoReferences: List<SeeAlsoReference>,
    override val javadocLink: String?,
    override val sourceLink: String?,
    val implementations: List<ImplementationIndex.Method>,
    val overriddenMethods: List<ImplementationIndex.Method>
) : CachedDoc {
    override val qualifiedName: String
        get() = "$className#$signature"
}